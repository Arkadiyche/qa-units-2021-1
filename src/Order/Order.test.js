import React from 'react'
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';
import Order from "./Order";
import {fakeOrders} from "../data/fakeOrders";

configure({adapter: new Adapter()});

getDate.mockReturnValue("29 ноября, ср, 2000 год");

describe('Order.js', () => {
  beforeEach(() => {
    jest.resetModules();
  });
  afterAll(() => {
    jest.resetModules();
  });

  it('without order', () => {
    const wrapper = shallow(<Order />);
    expect(wrapper.getElement()).toBeNull();
  });

  it('without shop and date', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.shop;
    delete order.date;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('without date', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.date;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('without shop', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.shop;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('without items', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.items;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('with all params', () => {
    const order = fakeOrders[0]
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
