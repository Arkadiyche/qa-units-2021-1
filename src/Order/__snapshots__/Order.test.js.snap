// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render snap with all 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      test
    </span>
    <span
      id="date"
    >
      29 ноября, ср, 2000 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      aa
    </div>
    <div
      className="Order-item"
      key="1"
    >
      ss
    </div>
    <div
      className="Order-item"
      key="2"
    >
      dd
    </div>
  </div>
</div>
`;

exports[`Order.js without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      29 ноября, ср, 2000 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;
